log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileIO.java

```diff
@@ -12,39 +12,46 @@
 public class FileIO {
     private String file_name;
     private String directory;
-    private static final Charset charset = Charset.forName( "UTF-8" );
+    private static final Charset charset = Charset.forName("UTF-8");
 
     /**
      * Constructor
+     *
      * @param directory Directory relative to running instance
      * @param file_name File name
      */
-    public FileIO( String directory, String file_name ) {
+    public FileIO(String directory, String file_name) {
         this.file_name = file_name;
         this.directory = directory;
     }
 
     /**
      * Creates file if it doesn't exist
+     *
      * @return Creation success
      */
-    public boolean createFile() {
+    public synchronized boolean createFile() {
```

Comment:
This one is a bit worrying. I can see there is a synchronization happening, but for the time being we have no explicit threading involved. Is this just a guard for the future? I.e. I think I see the point here- basically you are trying to make sure that the file is not created twice (or we dont' get that error).